File: python/hot100/44.二叉搜索树中第 K 小的元素.py
class Solution:
    def kthSmallest(self, root: Optional[TreeNode], k: int) -> int:
        if k<=0 or not root: return -1
        stack=[]
        count=0
        current=root
        while current or stack:
            while current:
                stack.append(current)
                current=current.left
            current=stack.pop()
            count+=1
            if count==k:
                return current.val
            current=current.right
        return -1

File: cpp/hot100/44.二叉搜索树中第 K 小的元素.cpp
class Solution {
public:
    int kthSmallest(TreeNode* root, int k) {
        if(!root) return -1;
        if(k<=0) return -1;
        stack<TreeNode*> stack;
        TreeNode* current=root;
        int count=0;
        while(current||!stack.empty()){
            while(current){
                stack.push(current);
                current=current->left;
            }
            current=stack.top();
            stack.pop();
            count++;
            if(count==k) return current->val;
            current=current->right;
        }
        return -1;
    }
};